src/polycap-out.c

```diff
@@ -8,15 +8,19 @@ void polycap_out(size_t n_energies, double *energies, double *efficiencies)
 {
```

Comment:
Ik ben geen fan van deze functie... Ik denk dat het beter is om `polycap_description_get_transmission_efficiencies` zodanig te herschrijven dat het een nieuwe struct `polycap_transmission_efficiencies` produceert met de resultaten en de oorspronkelijke energies array ipv een `int`. Iets zoals:

```c
// voor de private header
struct _polycap_transmission_efficiencies {
  size_t n_energies;
  double *energies;
  double *efficiencies;
};

// voor de publieke header
struct struct _polycap_transmission_efficiencies;
typedef struct _polycap_transmission_efficiencies polycap_transmission_efficiencies;

polycap_transmission_efficiencies* polycap_description_get_transmission_efficiencies(polycap_description *description, polycap_source *source, size_t n_energies, double *energies);

polycap_transmission_efficiencies_free(polycap_transmission_efficiencies *efficiencies);
polycap_transmission_efficiencies_write_hdf5(polycap_transmission_efficiencies *efficiencies, const char *filename);
// en nog andere functies bob om de gebruiker toegang te geven tot de efficiencies met een getter...
```

wat vindt je hiervan?

Reply:
Lijkt me goed. Lost direct ook enkele probleempjes op die ik had om bvb leak coordinates etc mee te geven naar output. Enige nadeel zal zijn dat ik dan de _polycap_transmission_efficiencies struct ook dien door te geven in enkele berekeningen, maar kzal zien of ik dat mss op een elegante manier kan circumventen.